java/maven.junit.ui/src/org/netbeans/modules/maven/junit/ui/MavenJUnitTestMethodNode.java

```diff
@@ -62,52 +63,71 @@ public MavenJUnitTestMethodNode(Testcase testcase, Project project, String proje
+            boolean integrationTest = "INTEGRATION".equals(getTestcase().getType()); // NOI18N
             Project suiteProject = FileOwnerQuery.getOwner(testFO);
             if (suiteProject != null) {
                 ActionProvider actionProvider = suiteProject.getLookup().lookup(ActionProvider.class);
                 if (actionProvider != null) {
-                    boolean runSupported = false;
-                    boolean debugSupported = false;
-                    for (String action : actionProvider.getSupportedActions()) {
-                        if (!runSupported && action.equals(COMMAND_RUN_SINGLE_METHOD)) {
-                            runSupported = true;
-                            if (debugSupported) {
-                                break;
-                            }
-                        }
-                        if (!debugSupported && action.equals(COMMAND_DEBUG_SINGLE_METHOD)) {
-                            debugSupported = true;
-                            if (runSupported) {
-                                break;
-                            }
-                        }
-                    }
-
                     SingleMethod methodSpec = new SingleMethod(testFO, testcase.getName());
                     Lookup nodeContext = Lookups.singleton(methodSpec);
-                    if (runSupported && actionProvider.isActionEnabled(COMMAND_RUN_SINGLE_METHOD,
-                            nodeContext)) {
-                        actions.add(new TestMethodNodeAction(actionProvider,
+
+                    for (String action : actionProvider.getSupportedActions()) {
+                        if (unitTest
+                            && action.equals(COMMAND_RUN_SINGLE_METHOD)
+                            && actionProvider.isActionEnabled(action, nodeContext)) {
+                            actions.add(new TestMethodNodeAction(actionProvider,
                                 nodeContext,
                                 COMMAND_RUN_SINGLE_METHOD,
                                 Bundle.LBL_RerunTest()));
-                    }
-                    if (debugSupported && actionProvider.isActionEnabled(COMMAND_DEBUG_SINGLE_METHOD,
-                            nodeContext)) {
-                        actions.add(new TestMethodNodeAction(actionProvider,
+                        }
+                        if (unitTest
+                            && action.equals(COMMAND_DEBUG_SINGLE_METHOD)
+                            && actionProvider.isActionEnabled(action, nodeContext)) {
+                            actions.add(new TestMethodNodeAction(actionProvider,
                                 nodeContext,
                                 COMMAND_DEBUG_SINGLE_METHOD,
                                 Bundle.LBL_DebugTest()));
+                        }
+                        if (integrationTest
+                            && action.equals("integration-test.single")
```

Comment:
Missing 4 `NOI18N`s

Maybe, should add `NOI18N` to `aName = "";` and `bName = "";`